.github/workflows/run-tests.yml

```diff
@@ -154,7 +154,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Upload event file artifact
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
```

Comment:
Should this be v4? That seems to be the latest version: https://github.com/actions/upload-artifact

However, v3 should be fine as well, just maybe less future proof. FWIW it looks like there are breaking change in v4 that might need to be consdiered if we upgrade that far: https://github.com/actions/upload-artifact/tree/main?tab=readme-ov-file#breaking-changes

Whereas no breaking changes are mentioned for v3: https://github.com/actions/upload-artifact/tree/v3

Also could you try temporarily making this change in your PR and see if it addresses the CI issue?

Reply:
I don't _think_ it will cause an issue, depending on what the scope of "same name" is for artifacts.

Reply:
Yeah v4 seems like it would be a bigger lift, and we use v3 all over the place already so this is more consistent. Also, see that the change fixed the PR [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1563).